### ISSUE TITLE
Bug: typegen resolves "exit" actions to 'xstate.init'

### ISSUE THREAD

**Issue by SimeonC** [statelyai/xstate-tools#89]
### Description

Running typegen on this code, the exit event is incorrectly typed to 'xstate.init' as in this screenshot.

![CleanShot 2022-03-02 at 11 11 31](https://user-images.githubusercontent.com/1085899/156281571-a7544572-5d5c-4722-8450-9924356c9733.png)

### Expected result

Exit events should be at least typed to "all events" in the machine `schema: { events }`.

In this particular case it would have been nice to have it typed to `FOCUS` event, but that's probably impractical.

### Actual result

Exit actions are typed to `'xstate.init'` instead of `''`

### Reproduction

NA - see additional context for example machine and output

### Additional context

```ts
const test =
  /** @xstate-layout N4IgpgJg5mDOIC5gF8A0IB2B7CdGgAoBbAQwGMALASwzAEoA6AMyoCdYAXfEABy1iocqWDNwAeiACwAmdAE9EATgCMDAAwa1kgMw7pa6ZMWTkaEMXLVadbnwFCR4xMoAc85y-WaX06QFYAdgA2X0VFaVNTIA */
  createMachine(
    {
      tsTypes: {} as import('./machine.typegen').Typegen0,
      schema: { events: {} as { type: 'SOME_ACTION' | 'B'; value: string } },
      id: '(machine)',
      states: {
        first: {
          exit: 'exitAction',
          on: { SOME_ACTION: 'second' }
        },
        second: {}
      }
    },
    {
      actions: {
        exitAction: (ctx, event) => {
          const t = event.value;
        }
      }
    }
  );
```

```ts
export interface Typegen0 {
  '@@xstate/typegen': true;
  eventsCausingActions: {
    exitAction: 'xstate.init';
  };
  internalEvents: {
    'xstate.init': { type: 'xstate.init' };
  };
  invokeSrcNameMap: {};
  missingImplementations: {
    actions: never;
    services: never;
    guards: never;
    delays: never;
  };
  eventsCausingServices: {};
  eventsCausingGuards: {};
  eventsCausingDelays: {};
  matchesStates: 'first' | 'second';
  tags: never;
}
```

**Comment by Andarist** [comment]
I've transferred this issue to https://github.com/statelyai/xstate-tools - since the algorithm for computing those events is living here.

**Comment by mattpocock** [comment]
Agree with all of the above, the algorithm should take this into account. This has been one of its weaknesses since it lived in xstate-codegen.

**Comment by mattpocock** [comment]
Added in #114, but need to co-ordinate it with XState's release of `xstate.stop`

**Comment by with-heart** [comment]
I think this can be closed. Merged in statelyai/xstate#3126 and I see it working in local typegen
